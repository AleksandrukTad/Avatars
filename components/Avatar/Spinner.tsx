import React from "react";

import { Circle, Spinner } from "./Spinner.style";

interface ILoadingProps extends React.Props<any> {
  loading: boolean;
}

const Loading: React.FunctionComponent<ILoadingProps> = ({ loading }) => {
  return (
    <Spinner loading={loading}>
      <Circle cx="20" cy="20" r="18" />
    </Spinner>
  );
};

export default Loading;
