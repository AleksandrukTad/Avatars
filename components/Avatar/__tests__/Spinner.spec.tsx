import "jest-dom/extend-expect";
import "jest-styled-components";

// __tests__/fetch.js
import { cleanup, render } from "@testing-library/react";
import React from "react";
// this adds custom jest matchers from jest-dom

import Spinner from "../Spinner";

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test("Render spinner with loading true", async () => {
  const { asFragment } = render(<Spinner loading={true} />);
  // you can also get a `DocumentFragment`, which is useful if you want to compare nodes across renders
  expect(asFragment()).toMatchSnapshot();
});

test("Render spinner with loading false", async () => {
  const { asFragment } = render(<Spinner loading={false} />);
  // you can also get a `DocumentFragment`, which is useful if you want to compare nodes across renders
  expect(asFragment()).toMatchSnapshot();
});
