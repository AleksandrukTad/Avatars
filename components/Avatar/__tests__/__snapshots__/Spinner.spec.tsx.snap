// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render spinner with loading false 1`] = `
<DocumentFragment>
  .c0 {
  display: none;
  position: absolute;
  width: 40px;
  height: 40px;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
  z-index: 99999;
}

.c1 {
  fill: transparent;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  stroke-width: 4;
  stroke-linecap: round;
  stroke-dasharray: 125.60000000000001;
  -webkit-transform-origin: 20px 20px 0px;
  -ms-transform-origin: 20px 20px 0px;
  transform-origin: 20px 20px 0px;
  -webkit-animation: kRGcnj 1s linear infinite;
  animation: kRGcnj 1s linear infinite;
  z-index: 99;
}

<svg
    class="c0"
  >
    <circle
      class="c1"
      cx="20"
      cy="20"
      r="18"
    />
  </svg>
</DocumentFragment>
`;

exports[`Render spinner with loading true 1`] = `
<DocumentFragment>
  .c0 {
  display: block;
  position: absolute;
  width: 40px;
  height: 40px;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
  z-index: 99999;
}

.c1 {
  fill: transparent;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  stroke-width: 4;
  stroke-linecap: round;
  stroke-dasharray: 125.60000000000001;
  -webkit-transform-origin: 20px 20px 0px;
  -ms-transform-origin: 20px 20px 0px;
  transform-origin: 20px 20px 0px;
  -webkit-animation: kRGcnj 1s linear infinite;
  animation: kRGcnj 1s linear infinite;
  z-index: 99;
}

<svg
    class="c0"
  >
    <circle
      class="c1"
      cx="20"
      cy="20"
      r="18"
    />
  </svg>
</DocumentFragment>
`;
